cpp/src/parquet/schema.cc

```diff
@@ -550,11 +550,11 @@ std::unique_ptr<Node> Unflatten(const format::SchemaElement* elements, int lengt
     int field_id = current_id++;
     const void* opaque_element = static_cast<const void*>(&element);
 
-    if (element.num_children == 0) {
+    if (!element.__isset.num_children) {
```

Comment:
should this be an or?

Reply:
Hmm, why would it be an or? If `num_children == 0`, this can still be a group node.

Reply:
Though it seems to produce failures on some Python Parquet tests (because of legacy files?). Let me see.